common/journald/src/stream.rs

```diff
@@ -208,28 +207,35 @@ impl Reader {
             Err(e) => return Err(JournalError::BadRead(e)),
         };
 
-        let timestamp = match self
+        match self
             .reader
             .timestamp()
-            .map(|timestamp| timestamp.duration_since(UNIX_EPOCH))
+            .ok()
+            .map(|timestamp| SystemTime::now().duration_since(timestamp).ok())
+            .flatten()
         {
-            Ok(Ok(timestamp)) => Local
-                .timestamp(timestamp.as_secs() as i64, 0)
-                .format("%b %d %H:%M:%S")
-                .to_string(),
-            Ok(Err(_)) | Err(_) => {
-                warn!("unable to read timestamp associated with journald record");
-                Local::now().format("%b %d %H:%M:%S").to_string()
+            Some(duration) => {
+                // Reject any records with a timestamp older than 30 seconds
+                if duration >= Duration::from_secs(30) {
+                    warn!("Received a stale journald record, reseeking pointer");
```

Comment:
Do you think this should be a warn or just an info?

Reply:
I think an info for now. Looking at https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html I think there might be some combination of fields we can use to do this more precisely, but I think this makes sense until we add something like lookback for journald

Reply:
Updated to info!